src/components/ProgressIndicator/ProgressIndicator.jsx

```diff
@@ -1,141 +1,364 @@
-    items,
-    currentItemId,
-  ]);
+  useEffect(() => setCurrentStep(currentItemId), [currentItemId]);
 
-  // Only recalculate current step if inputs change
-  const currentStep = matchingIndex > -1 ? matchingIndex : 0;
-  return (
-    <StyledProgressIndicator
-      className={classnames(className, `${iotPrefix}--progress-indicator`)}
-      onChange={handleChange}
-      currentIndex={currentStep}
-      isVerticalMode={isVerticalMode}
-      vertical={isVerticalMode}
+  const classes = classnames({
+    [`${iotPrefix}--progress-indicator`]: true,
+    [`${iotPrefix}--progress-indicator--vertical`]: isVerticalMode,
+    [className]: className,
+  });
+
+  return newItems.length > 1 ? (
+    <ul
+      className={classes}
+      data-testid={`${iotPrefix}--progress-indicator-testid`}
+      onChange={onClickItem || handleChange}
     >
-      {items.map(({ id, label, secondaryLabel, description }) => (
-        <StyledProgressStep
-          key={id}
-          label={label}
-          data-label={label}
-          secondaryLabel={secondaryLabel}
-          description={description || label}
-          showLabel={showLabels || currentItemId === id}
-          stepWidth={stepWidth}
-          isVerticalMode={isVerticalMode}
-        />
-      ))}
-    </StyledProgressIndicator>
-  );
+      {newItems.map(
+        (
+          { id, label, secondaryLabel, description, disabled, invalid, stepNumber, level },
+          index
+        ) => (
+          <ProgressStep
+            id={id}
+            key={id}
+            label={label}
+            secondaryLabel={secondaryLabel}
+            description={description || label}
+            current={currentStep === id}
+            complete={getCurrentIndex() > index}
+            incomplete={getCurrentIndex() < index}
+            mainStep={level === 0}
+            subStep={level > 0}
+            onChange={handleChange}
+            stepNumber={stepNumber}
+            vertical={isVerticalMode}
+            showLabel={showLabels}
+            stepWidth={stepWidth}
+            lastItem={lastItemId() === id}
+            disabled={disabled}
+            invalid={invalid}
+            isClickable={isClickable}
+          />
+        )
+      )}
+    </ul>
+  ) : null;
 };
 
-ProgressIndicator.propTypes = propTypes;
-ProgressIndicator.defaultProps = defaultProps;
+ProgressIndicator.propTypes = {
+  className: PropTypes.string,
+  items: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: IDPropTypes,
+      label: PropTypes.string.isRequired,
+      secondaryLabel: PropTypes.string,
+      description: PropTypes.string,
+      disabled: PropTypes.bool,
+      invalid: PropTypes.bool,
+    })
+  ),
+  currentItemId: IDPropTypes,
+  showLabels: PropTypes.bool,
+  stepWidth: PropTypes.number,
+  isVerticalMode: PropTypes.bool,
+  isClickable: PropTypes.bool,
+  setStep: PropTypes.func,
+  onClickItem: PropTypes.func,
```

Comment:
I believe this prop is effectively deprecated, right? Could we add a deprecation warning to this component about this prop?

We have a helper for it:

https://github.com/IBM/carbon-addons-iot-react/blob/4d14293cc189abf9d222a11d0c5327cc347d0e00/src/components/PageTitleBar/PageTitleBar.jsx#L18-L21

Reply:
I have added the onClickItem prop back. No need to deprecate it. I made some minor refactoring.